lib/reversetunnel/srv.go

```diff
@@ -1231,7 +1231,7 @@ func newRemoteSite(srv *server, domainName string, sconn ssh.Conn) (*remoteSite,
 	// certificate cache is created in each site (instead of creating it in
 	// reversetunnel.server and passing it along) so that the host certificate
 	// is signed by the correct certificate authority.
-	certificateCache, err := newHostCertificateCache(srv.localAuthClient)
+	certificateCache, err := newHostCertificateCache(remoteSite.remoteClient)
```

Comment:
Could you please explain what this change means in practice? Is it the core of the fix? In this scenario, what is the role of the client object (not explained in the docs), and how does changing it fix the issue?